client/rest/src/main/java/org/elasticsearch/client/DeadHostState.java

```diff
@@ -96,19 +84,4 @@ public String toString() {
                 ", deadUntilNanos=" + deadUntilNanos +
                 '}';
     }
-
-    /**
-     * Time supplier that makes timing aspects pluggable to ease testing
-     */
-    interface TimeSupplier {
-
-        TimeSupplier DEFAULT = new TimeSupplier() {
-            @Override
-            public long nanoTime() {
-                return System.nanoTime();
-            }
-        };
-
-        long nanoTime();
-    }
```

Comment:
I suppose these changes are gone due to a bad merge. I think this was a good improvement and I don't see a replacement for it and the corresponding tests in this PR.

Reply:
would be nice to use the fact that DeadHostState is now Comparable and the shallBeRetried method was convenient too I think.

Reply:
I think I see what you are getting at. We can keep `shallbeRetried` if we keep the `DeadHostState` and sort on it. That'd be fine.